### DISCUSSION TITLE
[MTR05] LIOJ1004 問題

### DISCUSSION THREAD

**Discussion by lucktanya33** [Lidemy/forum#73] [程式導師實驗計畫] [answered]
大家好，同上一篇也想問 LIOJ1004 **聯誼順序比大小** 解不出來的問題...目前使用題目上面的測試資料在本地端沒有問題。有爬文看自我檢討知道了這題有一個要注意的點是數字的範圍，因此使用 `BigInt` ，但在 LIOJ 上面還是會 Wrong Answer。不曉得有沒有人可以指點解題方向～～或是有什麼明顯的錯誤是目前沒有注意到的 謝謝！

題目

> 輸入第一行會是一個數字 M，1 &lt;= M &lt;= 501<=M<=50，代表一共有幾組比賽的結果。接著每一行會有三個用空白分隔的數字 A, B, K。A 代表 A 選擇的數字，B 代表 B 選擇的數字，兩者皆保證為正整數要特別注意的是 A 與 B 可能是很大的數字，但保證長度為 512 個位數以內。K 只會有兩種情況：1 或是 -1，若是 1 代表數字大的獲勝，K 若是 -1 代表數字小的獲勝

程式邏輯

> 先決定是比大或比小，再依據 A勝、B勝或平手的三種狀況輸出相對應的值

``` js
function solve(lines) {
let arrLength = Number(lines[0]) //先取出第一行 Line0
for(let i=1; i<=arrLength; i++){
    let [a,b,k] = lines[i].split(' ') // ['1', '2', '1']把每一行變成一個三個數字的陣列，每次取出
    //console.log([a,b,k]) 檢查測試資料
    if (k==1){ //k==1 代表班長決定比大
        if(BigInt(a) > BigInt(b)){
            console.log("A")
        }else if(BigInt(a)===BigInt(b)){
            console.log("Draw")
        }else{console.log("B")
        }
    } else{   //k!==1 代表班長決定比小
        if(BigInt(a) < BigInt(b)){
            console.log("A")
        }else if(BigInt(a)===BigInt(b)){
            console.log("Draw")
        }else{console.log("B")
    }
    }  
}
}
```

**Comment by aszx87410** [comment, marked as answer]
題目要求：

> 若是 A 贏請輸出 A，B 贏請輸出 B，平手則輸出 DRAW

**Reply by lucktanya33** [reply]
了解了，謝謝老師 ！
